src/events/api/serializers.py

```diff
@@ -11,117 +11,168 @@ class PrimarySpeakerSerializer(serializers.Serializer):
     github_profile_url = serializers.CharField()
     twitter_profile_url = serializers.CharField()
     facebook_profile_url = serializers.CharField()
+    bio = serializers.CharField()
 
 
-class TalkDetailSerializer(serializers.ModelSerializer):
+def format_speakers_data(request, speakers, show_details=False):
+    formatted = []
+    for s in speakers:
+        thumbnail_absolute_uri = request.build_absolute_uri(s.get_thumbnail_url())
+        data = {
+            'thumbnail_url': thumbnail_absolute_uri,
+            'name': s.speaker_name,
+        }
+        if show_details:
+            data = {
+                **data,
+                'bio': s.bio,
+                'github_profile_url': s.github_profile_url,
+                'twitter_profile_url': s.twitter_profile_url,
+                'facebook_profile_url': s.facebook_profile_url,
+            }
+        serialized = PrimarySpeakerSerializer(data=data).get_initial()
+        formatted.append(ReturnDict(serialized, serializer=PrimarySpeakerSerializer))
+    return formatted
+
+
+def flatten_proposal_field(representation):
+    """
+    The helper function that used in `to_representation()` for
+    flattening the `proposal` object from serialized
+    `ProposedTalkEvent` or `ProposedTutorialEvent`.
+    """
+    proposal_repr = representation.pop('proposal')
+    for key in proposal_repr:
+        representation[key] = proposal_repr[key]
+    return representation
+
+
+class TalkProposalSerializer(serializers.ModelSerializer):
     speakers = serializers.SerializerMethodField()
+    is_sponsored = serializers.ReadOnlyField(default=False)
 
     def get_speakers(self, obj):
-        return [
-            ReturnDict(PrimarySpeakerSerializer(
-                data={'thumbnail_url': i.user.get_thumbnail_url(),
-                      'name': i.user.speaker_name,
-                      'github_profile_url': i.user.github_profile_url,
-                      'twitter_profile_url': i.user.twitter_profile_url,
-                      'facebook_profile_url': i.user.facebook_profile_url}).get_initial(),
-                serializer=PrimarySpeakerSerializer) for i in obj.speakers]
+        request = self.context.get('request')
+        users = [s.user for s in obj.speakers]
```

Comment:
I think it is more readable to replace the parameter name from users to speakers.